### PR TITLE
[4][com_scheduler] Manual Execution Rule blocks Lazy and Webcron

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #43767 .

### Testing Instructions
see #43767

### Actual result BEFORE applying this Pull Request

no task is executed if there is one Manual

### Expected result AFTER applying this Pull Request

task is executed even if there is a Manual task

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by brbrbr** [comment]
I was wondering, wouldn't it be safer for future changes to clone the lockQuery

$idQuery= clone $lockQuery; // and move the $lockQuery->update part down.

This ensures that the conditions for idquery are always the same. And saves some code.

And I got a different issue while testing. If you trash a locked/running task, it blocks execution, since  $lockCountQuery does not ignore trashed tasks.

**Comment by alikon** [comment]
> And I got a different issue while testing. If you trash a locked/running task, it blocks execution, since $lockCountQuery does not ignore trashed tasks.

should be fixed 

@brbrbr can you test ?

File: administrator/components/com_scheduler/src/Model/TaskModel.php [hunk 1]
```diff
@@ -358,67 +358,105 @@ public function getItem($pk = null)
...
+        return $this->fetchTask($db, $now);
+    }
 
-            if ($runningCount !== 0) {
-                $db->unlockTables();
+    /**
+     * Checks if there are any running tasks in the database.
+     *
+     * @param \JDatabaseDriver $db The database driver to use.
+     * @return bool True if there are running tasks, false otherwise.
```
Comment (Quy) [inline administrator/components/com_scheduler/src/Model/TaskModel.php:413]: Add `@since` to these new methods.

**Comment by brbrbr** [comment]
Hi,

I went through the different scenarios and all seems to work fine now.

Nice work, you reshuffled quite a bit code, don't like the 'clone' :)

A bit beyond this fix; Locking remains an issue. If a task crashes the whole cue get's locked. And i can't even force a run (or run an unlock task) since neither the CLI nor the WebCron pass the allowConcurrent option.

I have some thought about that: https://github.com/joomla/joomla-cms/discussions/43775

File: administrator/components/com_scheduler/src/Model/TaskModel.php (hunk 1 above)
Comment (alikon) [reply administrator/components/com_scheduler/src/Model/TaskModel.php:413]: should be OK despite drone....

**Comment by softforge** [comment]
I have tested this item :white_check_mark: successfully on 1737666d450502aa236f460829d15aad3101b1ca<br /><br />Works well, good catch<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43784">issues.joomla.org/tracker/joomla-cms/43784</a>.</sub>

**Comment by mabdelaziz77** [comment]
I have tested this item :white_check_mark: successfully on 1737666d450502aa236f460829d15aad3101b1ca<br /><br />It works for me.
I have setup 1 manual task, 1 every minute and 1 every 2 minutes.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43784">issues.joomla.org/tracker/joomla-cms/43784</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43784">issues.joomla.org/tracker/joomla-cms/43784</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**